thunder/torch/langctx.py

```diff
@@ -46,9 +46,9 @@ def get_method(self, id: str, *args, **kwargs) -> Callable:
                 return prop(inps[0])
 
         # has_tensor_input is False
-        # Defers to the primitive language context when there are no tensor inputs=
-        #   (the primitive language context handles operations on numbers)
-        primsctx: LanguageContext = resolve_language(Languages.PRIMS)
+        # Defers to the CLANG language context when there are no tensor inputs=
+        #   (the clang language context handles operations on NumberProxies and Numbers)
+        primsctx: LanguageContext = resolve_language(Languages.CLANG)
```

Comment:
similarly, instead of PRIMs. dispatch through CLANG so we have a chance to handle type promotion for number proxies.